lib/Interpreter/TransactionPool.h

```diff
@@ -52,7 +57,7 @@ namespace cling {
       return T;
     }
 
-    void releaseTransaction(Transaction* T, bool reuse = true) {
+    void releaseTransaction(Transaction* T, bool reuse = !kDebugMode) {
```

Comment:
For my brain, that makes the logic more complex than needed: this setting should override the value passed as `reuse`  if `reuse` is `true` and `NDEBUG` is off. I.e. in `#ifndef NDEBUG`, `reuse` shall be `false`, whatever the argument passed. With your patch I'll need to check that no caller calls it with an explicit `true`, now or in the future.

Can we "hard-wire" this by overriding the value of `reuse = false` in `releaseTransaction`'s body for `#ifndef NDEBUG`?